source/plugin/data/keys.rst

```diff
@@ -0,0 +1,109 @@
+==========
+Using Keys
+==========
+
+Getting and offering data using a key
+=====================================
+
+Since accessing even a single point of data going from ``DataHolder`` to the correct ``DataManipulator`` tended to
+be quite tedious and bloat both the code and the import list, a more direct way of accessing values via ``Key``\ s
+was devised. Let's just start out with an example.
+
+**Code Example: Healing a data holder, if possible**
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.DataHolder;
+    import org.spongepowered.api.data.key.Keys;
+
+    public void heal(DataHolder target) {
+        if (target.supports(Keys.HEALTH)) {
+            double maxHealth = target.getOrNull(Keys.MAX_HEALTH);
+            target.offer(Keys.HEALTH, maxHealth);
+        }
+    }
+
+Now for the details of the above function.
+
+The first line checks if our given data holder possesses health. Only if he does, he can be healed after all.
+Since a data holder can not have health without having a maximum health and vice versa, a check for one of the
+keys using the ``supports()`` method suffices.
+
+The second line uses the ``getOrNull()`` function to ask the data holder for its maximum health. Besides
+``getOrNull()``, the methods ``get()`` and ``getOrElse()`` exist, all of which accept a ``Key`` as their first
+parameter. Generally, ``get()`` should be used. It will return an ``Optional`` of the data requested or
+``Optional.absent()`` if the data holder does not support the supplied key. The ``getOrNull()`` method we used is
+no more than a shortcut for ``get(key).orNull()``. In our example it is safe to use ``getOrNull()`` since we
+already verified that the value will be present in the first line and therefore ``getOrNull()`` will just relieve
+us of the ``Optional``. The third possibility would be the ``getOrElse()``, which accepts a default value as a
+second parameter to be returned if the value is not present on the data holder.
+
+In the third line, we offer data back to the data holder. We provide a ``Key`` denoting the current health and the
+before acquired maximum health, thus healing the data holder to full health. There are a variety of ``offer()``
+methods accepting different parameter sets, all of which return a ``DataTransactionResult`` containing information
+if the offer was accepted. For now, we'll use the one accepting a ``Key`` and a corresponding value, but we will
+encounter more in the next pages. Since we already know that our offer of current health is accepted (as the data
+holder supports it), we can silently discard the result.
+
+Transforming Data
+=================
+
+Other than getting a value, modify it and offer it back, there is another way of modifying data. Using a data
```

Comment:
"Other than getting a value, modify it and offer it back" --> "Other than getting, modifying, and offering a value"